File: sdk/go/data_fabric.go
package llmengine

import (
	"encoding/json"
	"fmt"
	"net/http"
)

type DataFabricClient struct {
	BaseURL    string
	HTTPClient *http.Client
}

func NewDataFabricClient(baseURL string) *DataFabricClient {
	return &DataFabricClient{
		BaseURL:    baseURL,
		HTTPClient: &http.Client{},
	}
}

func (c *DataFabricClient) ListDataSources() (map[string]interface{}, error) {
	resp, err := c.HTTPClient.Get(fmt.Sprintf("%s/v1/data-sources", c.BaseURL))
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var result map[string]interface{}
	if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
		return nil, err
	}
	return result, nil
}

File: sdk/typescript/src/data-fabric.ts
/**
 * TypeScript SDK Client for Data Fabric API.
 */

export class DataFabricClient {
  private baseUrl: string;

  constructor(baseUrl: string = "http://localhost:8000") {
    this.baseUrl = baseUrl.replace(/\/$/, "");
  }

  async listDataSources(): Promise<Record<string, any>> {
    const res = await fetch(`${this.baseUrl}/v1/data-sources`);
    if (!res.ok) throw new Error(`Fetch failed: ${res.statusText}`);
    return res.json();
  }

  async listCatalog(): Promise<Record<string, any>> {
    const res = await fetch(`${this.baseUrl}/v1/data-catalog`);
    if (!res.ok) throw new Error(`Fetch failed: ${res.statusText}`);
    return res.json();
  }
}
